py-polars/polars/series/series.py

```diff
@@ -4511,16 +4517,19 @@ def to_pandas(  # noqa: D417
                     else ""
                 )
 
-        pd_series = (
-            self.to_arrow().to_pandas(
+        pa_arr = self.to_arrow()
+        date_as_object = kwargs.pop("date_as_object", False)
+        if use_pyarrow_extension_array:
+            pd_series = pa_arr.to_pandas(
                 self_destruct=True,
                 split_blocks=True,
                 types_mapper=lambda pa_dtype: pd.ArrowDtype(pa_dtype),
+                date_as_object=date_as_object,
```

Comment:
is this necessary, given that there's already the `types_mapper` on the line above?

Reply:
I'll double-check!